submissions/VBystrov/tiny-world-oop/index.js

```diff
@@ -0,0 +1,110 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/VBystrov/a-tiny-JS-world
+   Web app: https://vbystrov.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Creature {
+  constructor(name, gender = 'unknown') {
+    this.name = name;
+    this.gender = gender;
+    this.friends = [];
+  }
+
+  addFriends(...newFriends) {
+    newFriends.forEach(({ name }) => this.friends.push(name));
+    return this.friends;
+  }
+}
+
+class Dog extends Creature {
+  constructor(name, gender) {
+    super(name, gender);
+    this.species = 'dog';
+    this.legs = 4;
+    this.saying = 'woof!';
```

Comment:
All of the creatures are of  certain species, have legs and a saying.
All shared props should be generalized into a base class.
Base class constructor can have different specification (number and set of arguments) than any of its child classes.
Let a child class constructor take arguments minimally required to set up an instance and call `super` with parameters required to get initialized via parent's constructor.

Reply:
Done.